tests/integration/test_dbapi_integration.py

```diff
@@ -171,22 +171,23 @@ def test_execute_many_select(trino_connection):
     assert "Query must return update type" in str(e.value)
 
 
-@pytest.mark.parametrize("connection_experimental_python_types,cursor_experimental_python_types,expected",
-                         [
-                             (None, None, False),
-                             (None, False, False),
-                             (None, True, True),
-                             (False, None, False),
-                             (False, False, False),
-                             (False, True, True),
-                             (True, None, True),
-                             (True, False, False),
-                             (True, True, True),
-                         ])
-def test_experimental_python_types_with_connection_and_cursor(
-        connection_experimental_python_types,
-        cursor_experimental_python_types,
-        expected,
+@pytest.mark.parametrize(
+    "connection_legacy_primitive_types,cursor_legacy_primitive_types,expected_legacy_primitive_types",
+    [
+        (None, None, False),
+        (None, False, False),
+        (None, True, True),
+        (False, None, False),
+        (False, False, False),
+        (False, True, True),
+        (True, None, True),
+        (True, False, False),
+        (True, True, True),
+    ])
```

Comment:
```suggestion
booleans = [None, True, False]

@pytest.mark.parametrize("connection_legacy_primitive_types", booleans)
@pytest.mark.parametrize("cursor_legacy_primitive_types", booleans)
@pytest.mark.parametrize("expected_legacy_primitive_types", booleans)
```

This results in cartesian product of all parameters being used.

Reply:
but, do we want to receive cartesian product and check every possible combination?

In my opinion we don't want to, because for example, let's have such case:
`(None, None, False)`
It means, that if we are not passing legacy_primitive_types to connection, nor to cursor, then expected value of this parameter is False, because this is the default.

Then, checking such combination:
`(None, None, True)`
is not needed, because such case doesn't exist.

Please correct me if I'm missing something.

Reply:
Can skip those cases within the test method. But I don't think those are bad cases to test (except None, None, True).

Reply:
I don't agree actually, there is nothing being skipped. Only the first 2 columns are inputs (with each having 3 possible values). the third value is the expected value. There are `3*3` values to test. I think the matrix is actually better than having to provide some code within the test to skip the wrong occurrences of the expected value.

Reply:
I have one more idea, we could parametrize only two first columns, and calculate 3rd column based on them, something like:


```python
booleans = [None, True, False]
@pytest.mark.parametrize("connection_legacy_primitive_types", booleans)
@pytest.mark.parametrize("cursor_legacy_primitive_types", booleans)
def test_legacy_primitive_types_with_connection_and_cursor(
...


# If legacy_primitive_types is passed to cursor, take value from it.
# If not, take value from legacy_primitive_types passed to connection.
# If legacy_primitive_types is not passed to cursor nor connection, default to False
if cursor_legacy_primitive_types is not None:
    expected_legacy_primitive_types = cursor_legacy_primitive_types
elif connection_legacy_primitive_types is not None:
    expected_legacy_primitive_types = connection_legacy_primitive_types
else:
    expected_legacy_primitive_types = False



```

So we could get rid of matrix, but incorporate some logic into test. I'm not sure what do you think is better approach?

Reply:
@damian3031 : Let's go that!

Reply:
@mdesmet @hashhar changed accordingly to snippet above, PTAL here.